fw.lua

```diff
@@ -325,7 +304,69 @@ local function _log_event(self, request_client, request_uri, rule, match)
 		end
 	}
 
-	lookup[self._event_log_target](t)
+	lookup[self._event_log_target](log_event)
+end
+
+-- constucts the log entry, and pushes it to the configured logging target
+local function _log_request(self, ctx, collections)
+	if (table.getn(ctx.matched_rules) ~= 0) then
+		local t = {
+			timestamp = ngx.time(),
+			client = collections["IP"],
+			method = collections["METHOD"],
+			uri = collections["URI"],
+			alert = ctx.matched_rules,
+			score = ctx.score
+		}
+
+		if self._event_log_request_arguments then
+			t.uri_args = ngx.var.request_uri
+		end
+
+		if self._event_log_request_headers then
+			t.request_headers = {}
+			-- name-value structure: this allows a mapping definition in document stores
+			for key, value in pairs(collections["HEADERS"]) do
+				local h = {
+					name = key,
+					value = value
+				}
+				table.insert(t.request_headers, h)
+			end
+		end
+
+		if (table.getn(self._event_log_ngx_vars) ~= 0) then
+			t.ngx = {}
+			for _, k in ipairs(self._event_log_ngx_vars) do
+				t.ngx[k] = ngx.var[k]
+			end
+		end
```

Comment:
same question about the structure here wrt. this collection vs. the request headers- why are the request headers treated different than this data?